File: ReadingSourceCode/JSPatchNotes/SCPatch/SCPatch/ViewController.m
//
//  ViewController.m
//  SCPatch
//
//  Created by ShannonChen on 2018/5/2.
//  Copyright © 2018年 ShannonChen. All rights reserved.
//

#import "ViewController.h"
#import "JSEngine.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    
    [JSEngine evaluateJavaScriptString:@""];
}

+ (void)callObjC {
    NSLog(@"调用 ObjC");
}

@end

File: iOSTutorials/多线程/NSOperation/NSOperationExample/NSOperationExample/ViewController.swift
//
//  ViewController.swift
//  NSOperationExample
//
//  Created by ShannonChen on 2017/12/15.
//  Copyright © 2017年 ShannonChen. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        
        
        let operation_1 = CustomOperation.init(identifier: "1")
        let operation_2 = CustomOperation.init(identifier: "2")
        let operation_3 = CustomOperation.init(identifier: "3")
        
        operation_2.addDependency(operation_1)
        operation_3.addDependency(operation_2)
        
        let operationQueue = OperationQueue()
        operationQueue.addOperation(operation_1)
        operationQueue.addOperation(operation_2)
        operationQueue.addOperation(operation_3)
        
        
    }



}

